Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -93,8 +93,9 @@ public final class SwiftTargetBuildDescription {
     var moduleOutputPath: AbsolutePath {
         // If we're an executable and we're not allowing test targets to link against us, we hide the module.
         let allowLinkingAgainstExecutables = (buildParameters.triple.isDarwin() || self.buildParameters.triple
-            .isLinux() || self.buildParameters.triple.isWindows()) && self.toolsVersion >= .v5_5 && target.type != .macro
-        let dirPath = (target.type == .executable && !allowLinkingAgainstExecutables) ? self.tempsPath : self
+            .isLinux() || self.buildParameters.triple.isWindows()) && self.toolsVersion >= .v5_5
+        let supportsTestableExecutablesFeature = (target as? SwiftTarget)?.supportsTestableExecutablesFeature == true
+        let dirPath = (supportsTestableExecutablesFeature && !allowLinkingAgainstExecutables) ? self.tempsPath : self
```

Comment:
So this wasn't actually checking for `is SwiftTarget` before which means this check applied beyond the testable executables feature. I'm reverting this to be as before to avoid unnecessary changes.